src/plugins/fs.c

```diff
@@ -47,6 +47,51 @@
  * A plugin for operations with file systems
  */
 
+typedef enum {
+    BD_FS_RESIZE,
+    BD_FS_REPAIR,
+    BD_FS_CHECK
+} BDFsOpType;
+
+/**
+ * BDFSInfo:
+ * @type: filesystem identifier, must be present
+ * @check_util: required utility for consistency checking, "" if not needed and NULL for no support
+ * @repair_util: required utility for repair, "" if not needed and NULL for no support
+ * @resize_util: required utility for resize, "" if not needed and NULL for no support
+ * @resize_mode: resize availability flags, 0 if no support
+ */
+typedef struct BDFSInfo
+{
+    const gchar *type;
+    const gchar *check_util;
+    const gchar *repair_util;
+    const gchar *resize_util;
+    BDFsResizeFlags resize_mode;
+} BDFSInfo;
+
+const BDFSInfo fs_info[] = {
+    {"xfs", "xfs_db", "xfs_repair", "xfs_growfs", BD_FS_ONLINE_GROW | BD_FS_OFFLINE_GROW},
+    {"ext2", "e2fsck", "e2fsck", "resize2fs", BD_FS_ONLINE_GROW | BD_FS_OFFLINE_GROW | BD_FS_OFFLINE_SHRINK},
+    {"ext3", "e2fsck", "e2fsck", "resize2fs", BD_FS_ONLINE_GROW | BD_FS_OFFLINE_GROW | BD_FS_OFFLINE_SHRINK},
+    {"ext4", "e2fsck", "e2fsck", "resize2fs", BD_FS_ONLINE_GROW | BD_FS_OFFLINE_GROW | BD_FS_OFFLINE_SHRINK},
+    {"vfat", "fsck.vfat", "fsck.vfat", "", BD_FS_OFFLINE_GROW | BD_FS_OFFLINE_SHRINK},
+    {NULL, NULL, NULL, NULL, 0}
+};
+
+static const BDFSInfo *
+get_fs_info (const gchar *type)
+{
+    g_return_val_if_fail (type != NULL, NULL);
+
+    for (guint n = 0; fs_info[n].type != NULL; n++) {
```

Comment:
I noticed that, but it only happens on RHEL/CentOS7. Please keep this in place, I'll change how we compile libblockdev so that this works fine.